eox_nelp/third_party_auth/pipeline.py

```diff
@@ -65,3 +75,36 @@ def close_mismatch_session(request, *args, user=None, **kwargs):  # pylint: disa
         logout(request)
 
     return {}
+
+
+def safer_associate_username_by_uid(  # pylint: disable=unused-argument
+    backend, details, response, *args, user=None, **kwargs,
+):
+    """Pipeline to retrieve user if possible matching uid with the username of a user.
+    The uid is based in the configuration of the saml with the field `attr_user_permanent_id`:
+    https://github.com/python-social-auth/social-core/blob/master/social_core/backends/saml.py#L49
+
+    This is using the idp and the uid inspired in:
+    https://github.com/python-social-auth/social-core/blob/master/social_core/backends/saml.py#L296C23-L297
+    Raises:
+        EoxNelpAuthException: If someone tries to have staff or superuser permission using tpa.
+
+    Returns:
+        dict: Dict with user if matches, if not return None.
+    """
+    if user:
+        return None
```

Comment:
this should always return a dict, in this case that should be an empty dict since the pipeline [logic](https://github.com/python-social-auth/social-core/blob/master/social_core/backends/base.py#L119) will check the kind of instance

Reply:
I was thinking about this and why other pipes work using `None` in the return. 
https://github.com/eduNEXT/eox-tenant/blob/master/eox_tenant/pipeline.py#L25
https://github.com/python-social-auth/social-core/blob/master/social_core/pipeline/social_auth.py#L67

Then I found that there is an  [or-line](https://github.com/python-social-auth/social-core/blob/master/social_core/backends/base.py#L118
)  that allow use `None` changing it to `{}`.
![image](https://github.com/eduNEXT/eox-nelp/assets/51926076/5f227d23-268d-4f61-93fd-5d928d37dd9e)